src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -620,15 +621,31 @@ protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessa
                 return Task.FromCanceled<HttpResponseMessage>(cancellationToken);
             }
 
-            HttpMessageHandlerStage handler = _handler ?? SetupHandlerChain();
-
             Exception? error = ValidateAndNormalizeRequest(request);
             if (error != null)
             {
                 return Task.FromException<HttpResponseMessage>(error);
             }
 
-            return handler.SendAsync(request, cancellationToken);
+            // SetupHandlerChain may block for a few seconds in some environments.
+            // (See https://github.com/dotnet/runtime/issues/115301. )
+            // The setup procedure is being enqueued to thread pool to prevent the caller from blocking
```

Comment:
```suggestion
            // SetupHandlerChain may block for a few seconds in some environments.
            // E.g. during the first access of HttpClient.DefaultProxy - https://github.com/dotnet/runtime/issues/115301.
            // The setup procedure is enqueued to thread pool to prevent the caller from blocking.
```